### DISCUSSION TITLE
Como encender leds de una cadena de max7219

### DISCUSSION THREAD

**Discussion by Veroledez** [tttapa/Control-Surface#529] [General]
Hola, gracias por tu trabajo en esta gran biblioteca.

Bien, el problema esta en que no se como encender del led 65 en adelante. 
Teniendo en cuenta lo siguiente:

MAX7219<2> max = SS;

Como emcenderia los leds del segundo max, probe declarando digitalWritte( max[1].pin(5), HIGH) y no compila. Podrias orientarme a como se accede a los otros max.

Muchas gracias

**Comment by EvgeniySavkin** [comment]
Greetings! Peter always asks for the full code to solve the problem. And why do you have 65 leds? It seems that MAX7219 supports the connection of 64 leds. And the example specifies connections from 0 to 63. "0" is the first led, "63" is the last, that is, 64.

Have you tried using the example from the library???

https://github.com/tttapa/Control-Surface/blob/master/examples/Arduino-Helpers/Hardware/Extended%20Input%20%26%20Output/2.%20Output/2.%20MAX7219/1.MAX7219-Blink/1.MAX7219-Blink.ino

**Reply by Veroledez** [reply]
Hola, hago referencia como led 65 a modo de ejemplo como el primer led de un segundo max7219 conectado en cadena, compartiendo el mismo pin SS, en resumen, puedo encender solo los leds del primer max7219, pero no los demas. Muchas gracias

**Reply by EvgeniySavkin** [reply]
Publique su código completo

**Comment by Veroledez** [comment]
Este es el codigo




```cpp
#include <Arduino_Helpers.h> // Include the Arduino Helpers library.
#include <AH/Hardware/ExtendedInputOutput/MAX7219.hpp>

using namespace ExtIO; 

// declaracion para usar 5 integrados max7219
MAX7219<5> max7219 {SPI, SS};    // tengo entendido que el numero entre estos <> indica el numero de max7219 a utilizar

 

void setup() {
  max7219.begin();      
  }

void loop() {
  digitalWrite(max7219.pin(0);  , HIGH);   // este que es el primer led del primer max7219 funciona
  digitalWrite(max7219.pin(64);  , HIGH);  //  este que seria el primer led del segundo max7219 no funciona
  delay(500);

digitalWrite(max7219.pin(0);  , LOW);   //  Funciona
digitalWrite(max7219.pin(64);  , LOW);   // no funciona
delay(500);

}
```


cada max son 64 leds y 5 max7219, seria un total de 320 leds

cuando uso max7219.pin(64);   o en lugar del 64 un numero mayor y menor que 320 hay comportamientos extraños, ya que se encienden leds de 2 o 3 max7219 diferentes

Estoy usando modulos matrices 8x8, de los que ya vienen listos, solo para conectar y encadenar, ya probre cada uno de los modulos de forma individual con un bucle for para encender los leds uno por uno, con lo que comprobe que los modulos estan en perfecto estado. Estoy usando una fuente de 5v 10a para alimentar el circuito por lo que ese tompoco deberia ser el problema, los  modulos son encadenados formando una matriz de 8x40, y la conexion a arduino esta con cables buenos y cortos, probados con los 74hc595 con lo que verifique que no hay perdida de datos en esq parte, encadene 10 de estos.

Muchas gracias

**Reply by EvgeniySavkin** [reply]
Tiene signos adicionales en el bucle "pin(0);,". Eliminar el exceso de punto y coma en todas las líneas. Esto es incorrecto en términos de sintaxis.

Intenta usar este código:
```
#include <Arduino_Helpers.h> // Include the Arduino Helpers library.
#include <AH/Hardware/ExtendedInputOutput/MAX7219.hpp>

using namespace ExtIO;

// declaracion para usar 5 integrados max7219
MAX7219<5> max7219 {SPI, SS}; // tengo entendido que el numero entre estos <> indica el numero de max7219 a utilizar

const pin_t ledpin1 = max7219.pin(0);
const pin_t ledpin2 = max7219.pin(319);

void setup() {
  max7219.begin();
  pinMode(ledpin1, OUTPUT);
  pinMode(ledpin2, OUTPUT);
}

void loop() {
  digitalWrite(ledpin1, HIGH); // este que es el primer led del primer max7219 funciona
  digitalWrite(ledpin2, LOW); // este que seria el primer led del segundo max7219 no funciona
  delay(500);
  digitalWrite(ledpin1, LOW); // Funciona
  digitalWrite(ledpin2, HIGH); // no funciona
  delay(500);
}
```

**Comment by Veroledez** [comment]
Me equivoque al escribirlo, no tenia el codigo a la mano, probe  tu codigo  ya en casa y el primer led del primer max funciona, los leds del segundo al quinto max no funciona o funcionan de forma erronea encienden varios a la v de distintos max

**Reply by EvgeniySavkin** [reply]
mi Consejo, de principiante a principiante, tome una cantidad menor de código. y luego aumentar la complejidad y el número.
Con gran respeto de Rusia.

**Comment by tttapa** [comment]
I'm unable to reproduce this issue, after fixing the semicolons in your code and accounting for the fact I only have two MAX7219s to test with, it works as expected:
```cpp
#include <Arduino_Helpers.h> // Include the Arduino Helpers library.
#include <AH/Hardware/ExtendedInputOutput/MAX7219.hpp>

using namespace ExtIO; 

// declaracion para usar 5 integrados max7219
MAX7219<5> max7219 {SPI, 10};    // tengo entendido que el numero entre estos <> indica el numero de max7219 a utilizar

void setup() {
  max7219.begin();      
}

void loop() {
  digitalWrite(max7219.pin(64*3), HIGH);
  digitalWrite(max7219.pin(64*4), HIGH);
  delay(500);

  digitalWrite(max7219.pin(64*3), LOW);
  digitalWrite(max7219.pin(64*4), LOW);
  delay(500);
}
```
I've tested this using the current `master` branch and the `1.2.0` release, both work (1.2.0 requires you to omit the `SPI` argument to the `MAX7219` constructor, but other than that, the code is identical).

If you can't get it to work, please post:
- The exact version of Control Surface you have installed (git commit hash, git branch or release version)
- The exact code you're using, a complete copy-paste of the code you have uploaded to the Arduino
- The Arduino board you're using
- A wiring diagram and/or a clear picture of your wiring
